### PR TITLE
efi: constify several functions and enable more warnings

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

**Comment by keszybz** [comment]
LGTM.

**Comment by yuwata** [comment]
CentOS CI (CentOS8) CI failure is unrelated. @mrc0mmand Could you restart the job?

**Comment by bluca** [comment]
The CentOS tests succeeded, but they couldn't update the PR status, probably because of this morning's outage:

```
09:23:23 Setting status of 505ab9dda2147375a72cd1593c8c483a9c516faa to SUCCESS with url https://jenkins-systemd.apps.ocp.ci.centos.org/job/upstream-vagrant-archlinux/5038/ and message: 'build passed (ping @mrc0mmand for any questions)
09:23:23  '
09:23:23 Using context: CentOS CI (Arch Linux)
09:23:31 Could not update commit status of the Pull Request on GitHub.
09:23:31 org.kohsuke.github.HttpException: Server returned HTTP response code: 500, message: 'null' for URL: 
```

File: src/boot/efi/sha256.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include <endian.h>
```
Comment (poettering) [inline src/boot/efi/sha256.h:4]: that's a glibc header, no? this is EFI code, we cannot just include glibc headers here. what am i missing?

Comment (poettering) [reply src/boot/efi/sha256.h:4]: actually, this looks wrong entirely. `__BYTE_ORDER__` and `__ORDER_LITTLE_ENDIAN__` are defined by gcc itself, not glibc. i.e.

https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html#Common-Predefined-Macros

endian defines `__BYTE_ORDER`, i.e. the subtle __ suffix is not there...

Comment (yuwata) [reply src/boot/efi/sha256.h:4]: #19802.

File: src/boot/efi/boot.c [hunk 2]
```diff
@@ -521,7 +521,7 @@ static BOOLEAN menu_run(
         uefi_call_wrapper(ST->ConOut->SetAttribute, 2, ST->ConOut, EFI_LIGHTGRAY|EFI_BACKGROUND_BLACK);
 
         /* draw a single character to make ClearScreen work on some firmware */
-        uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, L" ");
+        uefi_call_wrapper(ST->ConOut->OutputString, 2, ST->ConOut, (CHAR16*) L" ");
```
Comment (poettering) [inline src/boot/efi/boot.c:524]: why drop the qualifiers here? shouldn't matter, no?

Comment (yuwata) [reply src/boot/efi/boot.c:524]: `OutputString()` takes `CHAR16 *` instead of `const CHAR16 *`. So, it is necessary to cast the string, unfortunately.

**Pull request merged**